arangod/IResearch/IResearchLinkCoordinator.cpp

```diff
@@ -83,6 +83,77 @@ IResearchLinkCoordinator::IResearchLinkCoordinator(
   _sparse = true;   // always sparse
 }
 
+Result IResearchLinkCoordinator::init(
+    velocypack::Slice definition,
+    IResearchLink::InitCallback const& initCallback) {
+  auto r = IResearchLink::init(definition, initCallback);
+  auto& metric = IResearchLink::collection()
+                     .vocbase()
+                     .server()
+                     .getFeature<metrics::ClusterMetricsFeature>();
+  std::function batchToCoordinator =
+      [](metrics::ClusterMetricsFeature::CoordinatorMetrics& metrics,
+         std::string_view name, velocypack::Slice labels,
+         velocypack::Slice value) {
+        auto& cell = std::get<uint64_t>(
+            metrics[metrics::ClusterMetricsFeature::MetricKey{
+                name, labels.stringView()}]);
+        cell += value.getUIntUnchecked();
+      };
+  metrics::ClusterMetricsFeature::ToPrometheus batchToPrometheus{
+      [](std::string& result, std::string_view name) {
+        result.append("# HELP ").append(name);
+        result.push_back(' ');
+        result.append("NO HELP");  // TODO(MBkkt) Should be help string
+        result.push_back('\n');
+        result.append("# TYPE ").append(name).append(" gauge\n");
+      },
+      [](std::string& result, std::string_view globals,
+         metrics::ClusterMetricsFeature::MetricKey const& key,
+         metrics::ClusterMetricsFeature::MetricValue const& value) {
+        result.append(key.first);
+        result.push_back('{');
+        result.append(globals);
+        if (!key.second.empty()) {
+          if (!globals.empty()) {
+            result.push_back(',');
+          }
+          result.append(key.second);
+        }
+        result.push_back('}');
+        result.append(std::to_string(std::get<uint64_t>(value)));
+        result.push_back('\n');
+      }};
+  metric.add(arangosearch_num_docs::kName, batchToCoordinator,
+             batchToPrometheus);
+  metric.add(arangosearch_num_live_docs::kName, batchToCoordinator,
+             batchToPrometheus);
+  metric.add(arangosearch_num_segments::kName, batchToCoordinator,
+             batchToPrometheus);
+  metric.add(arangosearch_num_files::kName, batchToCoordinator,
+             batchToPrometheus);
+  metric.add(arangosearch_index_size::kName, batchToCoordinator,
+             batchToPrometheus);
+  std::function gaugeToCoordinator =
+      [](metrics::ClusterMetricsFeature::CoordinatorMetrics& metrics,
+         std::string_view name, velocypack::Slice labels,
+         velocypack::Slice value) {
+        auto labelsStr = labels.stringView();
+        auto& cell = std::get<uint64_t>(
+            metrics[metrics::ClusterMetricsFeature::MetricKey{
+                name, labelsStr.substr(0, labelsStr.find(",shard=\""))}]);
```

Comment:
If `labelsStr` does not contain `",shard=\""`, then this will fail. We should guard against that.

Reply:
We know that it contains...

Reply:
tri_assert was added